### PR TITLE
Add lsm6dsl specific samples + change its dts binding

### PR DESCRIPTION
**Pull request by avisconti**
This PR is composed by 3 commits, all 3 somehow related to lsm6dsl sensor.

### CODE REVIEW

**Comment by avisconti** [comment]
@galak @MaureenHelm 
If you can also provide your inputs would be good!

File: samples/sensor/lsm6dsl/README.rst
Comment (dbkinder) [inline samples/sensor/lsm6dsl/README.rst]: Rendering is better without the leading space:
```suggestion
- LSM6DSL http://www.st.com/en/mems-and-sensors/lsm6dsl.html
```

**Review by dbkinder — CHANGES_REQUESTED** [review]
Thanks for the documentation, one little tweak.

File: samples/sensor/lsm6dsl/README.rst
Comment (avisconti) [reply samples/sensor/lsm6dsl/README.rst]: ack

**Review by erwango — COMMENTED** [review]
Seems in conflict with #11854. Can you have a check?

**Comment by avisconti** [comment]
@erwango 
> Seems in conflict with #11854. Can you have a check?

Yes, I already noticed it, as @galak added me in the review of #11854.
Actually, I have also to change the commit (just meged) for lis2dh as well have the one for lis3dh (currently under review #11814)

**Comment by avisconti** [comment]
I have repushed it on top of #11854. Now both i2c and spi yaml file are using same compatible
string. Moreover I took care of the inputs from @dbkinder on README doc file.

**Comment by galak** [comment]
Going to mark this DNM for now until #11854 goes in.

**Review by dbkinder — APPROVED** [review]
doc changes LGTM, thanks!

**Comment by galak** [comment]
Can you rebase now that we merged #11854

**Comment by avisconti** [comment]
> Can you rebase now that we merged #11854

done!

**Comment by avisconti** [comment]
recheck

**Comment by avisconti** [comment]
Am I the only one who is getting this error during CI?

---------------
L/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/out-3nd-pass/disco_l475_iot1/samples/net/wifi/test/zephyr -lgcc -Wl,--print-memory-usage -mthumb -nostdlib -static -no-pie --coverage -Wl,-X -Wl,-N -Wl,--gc-sections -Wl,--build-id=none -Wl,--orphan-handling=warn -mabi=aapcs && :
Memory region         Used Size  Region Size  %age Used
           FLASH:       81788 B         1 MB      7.80%
            SRAM:       32640 B        96 KB     33.20%
        IDT_LIST:         184 B         2 KB      8.98zephyr/libzephyr.a(eswifi_bus_spi.c.obj):(.data.eswifi_bus_ops_spi+0x0): multiple definition of `eswifi_bus_ops_spi'
zephyr/libzephyr.a(eswifi_core.c.obj):(.bss.eswifi_bus_ops_spi+0x0): first defined here
zephyr/libzephyr.a(eswifi_bus_spi.c.obj): In function `eswifi_spi_init':
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/drivers/wifi/eswifi/eswifi_bus_spi.c:101: multiple definition of `uarthost'
zephyr/libzephyr.a(eswifi_core.c.obj):/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/drivers/wifi/eswifi/eswifi_core.c:294: first defined here
zephyr/libzephyr.a(eswifi_offload.c.obj):(.bss.eswifi_bus_ops_spi+0x0): multiple definition of `eswifi_bus_ops_spi'
zephyr/libzephyr.a(eswifi_core.c.obj):/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/drivers/wifi/eswifi/eswifi_core.c:294: first defined here
zephyr/libzephyr.a(eswifi_offload.c.obj): In function `eswifi_off_accept':
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/drivers/wifi/eswifi/eswifi_offload.c:88: multiple definition of `uarthost'
zephyr/libzephyr.a(eswifi_core.c.obj):/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/drivers/wifi/eswifi/eswifi_core.c:294: first defined here
collect2: error: ld returned 1 exit status
%
ninja: build stopped: subcommand failed.
-----------------------------------

**Comment by galak** [comment]
> @galak
> Am I the only one who is getting this error during CI?

Nope, you aren't.  Looking to see how this happened.

**Comment by galak** [comment]
> Nope, you aren't. Looking to see how this happened.

Fix in PR #11988

**Comment by avisconti** [comment]
> Fix in PR #11988 

OK, I'll wait for the merge of that PR

**Comment by avisconti** [comment]
Merged!

**Review by galak — COMMENTED** [review]
Just noticed, and while not directly related CONFIG_LSM6DSL_BUS_TYPE we should be able to replace with {DT_ST_LSM6DSL_BUS_I2C and/or DT_ST_LSM6DSL_BUS_SPI} that are now generated.

**Review by galak — CHANGES_REQUESTED** [review]
The sample never gets built for an platform via sanitycheck:

./scripts/sanitycheck -T samples/sensor/lsm6dsl/

I wondering if we can wait for #11800 and than use a filter on DT_COMPAT_ST_LSM6DSL

**Comment by avisconti** [comment]
This PR has been rebased onto #12105 
Must wait til that one will be merged

**Comment by avisconti** [comment]
rebased after #12105 merging.
@galak 
I added DT_COMPAT_ST_LSM6DSL filter in sample.yaml, but still it seems sanitycheck is never
called for any platform. I'm not so familiar with this scripts, so I may need an help from you.

**Review by galak — CHANGES_REQUESTED** [review]
You need to add 'lsm6dsl' to the board yaml that support it.

**Comment by avisconti** [comment]
@galak 
> you need to add 'lsm6dsl' to the board yaml that support it.

removed the filter and added lsm6dsl to ArgonKey board.
Then re-based and re-pushed!
Thx for your support.

**Review by erwango — CHANGES_REQUESTED** [review]
2 minor comments, otherwise tested ok

File: boards/arm/96b_argonkey/96b_argonkey.yaml
Comment (avisconti) [reply boards/arm/96b_argonkey/96b_argonkey.yaml]: Sure.

File: samples/sensor/lsm6dsl/src/main.c
Comment (avisconti) [reply samples/sensor/lsm6dsl/src/main.c]: ack

**Comment by avisconti** [comment]
Modified according to @erwango observation and re-pushed.

**Review by erwango — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Pull request merged**
